com.unity.shadergraph/Editor/Importers/GraphUtil.cs

```diff
@@ -0,0 +1,69 @@
+using System.IO;
+using System.Text;
+using UnityEditor.ProjectWindowCallback;
+using UnityEngine;
+using UnityEngine.Rendering;
+
+namespace UnityEditor.ShaderGraph.GraphDelta
+{
+    public class NewGraphAction : EndNameEditAction
+    {
+        public override void Action(int instanceId, string pathName, string resourceFile)
+        {
+            ShaderGraphAssetHelper helper = ScriptableObject.CreateInstance<ShaderGraphAssetHelper>();
+            GraphHandler output = new GraphHandler();
+            var reg = Registry.Default.DefaultRegistry.CreateDefaultRegistry();
+            var contextKey = Registry.Registry.ResolveKey< Registry.Default.DefaultContext>();
+            output.AddContextNode(contextKey, reg);
+            helper.GraphDeltaJSON = output.ToSerializedFormat();
+            File.WriteAllText(pathName, EditorJsonUtility.ToJson(helper, true), Encoding.UTF8);
+            AssetDatabase.ImportAsset(pathName);
+            Selection.activeObject = AssetDatabase.LoadAssetAtPath<Shader>(pathName);
+        }
+    }
+
+    public static class GraphUtil
+    {
+
+        [MenuItem("Assets/Create/Shader Graph 2/Blank Graph", priority = CoreUtils.Priorities.assetsCreateShaderMenuPriority)]
```

Comment:
If this is the golden code path for creating properly configured graph assets now, we should either:
a) retire the other menu item we had for this (`Create->Shader Graph->Blank Shader Graph`), or replace what that code path did with this instead, for reference that code lives in `GraphAssetUtils`

b) provide a way to call this using the Onboarding Provider? (i.e. the blank Shader Graph window screen that offers to create a new asset)

Reply:
The Onboarding provider stuff lives in `ShaderGraphOnboardingProvider::CreateBlankShaderGraph` so replacing whats there with this code would be the way to go

Reply:
We were deliberately not overwriting old code paths, but @esmelusina if youre okay with this I can do it. Otherwise, option b would be easy I think

Reply:
Yea- don't touch anything in Current right now.

Does onboarding provider work with this? Eg. If I create an asset through the onboarding provider, does it behave?

Reply:
OnboardingProvider isn't old, it's new. It's the thing that offers to create a new asset when you access shadergraph through the `window -> shaders -> shadergraph` menu.

It's located in `GraphUI/ShaderGraphOnboardingProvider.cs`

Reply:
Oh, I thought I commented on this-- Sai is talking about the new Onboarding Provider in GraphUI, not the one for the current tool.